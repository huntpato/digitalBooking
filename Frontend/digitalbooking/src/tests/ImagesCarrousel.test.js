import React from 'react';
import { shallow } from 'enzyme';
import ImagesCarrousel from '../components/ImagesCarousel/ImagesCarousel'
import hotels from '../assets/hotels.png'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals'

jest.mock('react-router-dom', () => ({
    useParams: jest.fn().mockReturnValue({ nifUuid: 'nif123' }),
    useHistory: jest.fn(),
    useLocation:jest.fn().mockReturnValue({
      pathname: "/register",
      search: "",
      hash: "",
      state: null,
      key: "5nvxpbdafa",
    }),
  }));


describe('Pruebas en componente <ImagesCarrousel/>', ()=> {

    const wrapper= shallow(<ImagesCarrousel imagesGallery= {["../assets/hotels.png", "../assets/hotels.png", "../assets/hotels.png", "../assets/hotels.png", "../assets/hotels.png"]} />);

    test('Comprobar que <ImagesCarrousel/> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const main = wrapper.find('div').at(0);
        expect(main.hasClass('carouselContainer')).toBe(true);
    })
    
    /*test('Verificar el cierre del carrousel', () => {
      let mockFn = jest.fn();
      wrapper.instance().setGallery = mockFn;
      const x = wrapper.find('h5').props().onClick() ;
      expect(mockFn).toHaveBeenCalledTimes(1);
  })*/

})
