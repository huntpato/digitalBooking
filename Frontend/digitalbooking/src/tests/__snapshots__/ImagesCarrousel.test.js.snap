// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <ImagesCarrousel/> Comprobar que <ImagesCarrousel/> renderiza correctamente. 1`] = `
<div
  className="carouselContainer"
>
  <h5
    className="close"
    onClick={[Function]}
  >
    X
  </h5>
  <Carousel
    animationHandler="slide"
    autoPlay={true}
    axis="horizontal"
    centerSlidePercentage={80}
    infiniteLoop={false}
    interval={3000}
    labels={
      Object {
        "item": "slide item",
        "leftArrow": "previous slide / item",
        "rightArrow": "next slide / item",
      }
    }
    onChange={[Function]}
    onClickItem={[Function]}
    onClickThumb={[Function]}
    onSwipeEnd={[Function]}
    onSwipeMove={[Function]}
    onSwipeStart={[Function]}
    preventMovementUntilSwipeScrollTolerance={false}
    renderArrowNext={[Function]}
    renderArrowPrev={[Function]}
    renderIndicator={[Function]}
    renderItem={[Function]}
    renderThumbs={[Function]}
    selectedItem={0}
    showArrows={true}
    showIndicators={true}
    showStatus={true}
    showThumbs={true}
    statusFormatter={[Function]}
    stopOnHover={true}
    stopSwipingHandler={[Function]}
    swipeAnimationHandler={[Function]}
    swipeScrollTolerance={5}
    swipeable={true}
    thumbWidth={100}
    transitionTime={1400}
    verticalSwipe="standard"
    width="100%"
  >
    <div>
      <img
        alt="Imagen del producto"
        src="../assets/hotels.png"
      />
    </div>
    <div>
      <img
        alt="Imagen del producto"
        src="../assets/hotels.png"
      />
    </div>
    <div>
      <img
        alt="Imagen del producto"
        src="../assets/hotels.png"
      />
    </div>
    <div>
      <img
        alt="Imagen del producto"
        src="../assets/hotels.png"
      />
    </div>
    <div>
      <img
        alt="Imagen del producto"
        src="../assets/hotels.png"
      />
    </div>
  </Carousel>
</div>
`;
